components/go/pkg/api/client.go

```diff
@@ -0,0 +1,141 @@
+package api
+
+import (
+	"encoding/json"
+	"errors"
+	"io"
+	"log"
+	"net/http"
+	"strings"
+	"time"
+
+	config "github.com/chef/chef-licensing/components/go/pkg/config"
+)
+
+type apiResponse struct {
+	Data       bool   `json:"data"`
+	Message    string `json:"message"`
+	StatusCode int    `json:"status_code"`
+}
+
+type clientAPIResponse struct {
+	Data struct {
+		Client LicenseClient `json:"client"`
+	} `json:"data"`
+	Message    string `json:"message"`
+	StatusCode int    `json:"status_code"`
+}
+
+type LicenseClient struct {
+	LicenseType string `json:"license"`
+	Status      string `json:"status"`
+	ChangesTo   string `json:"changesTo"`
+	ChangesOn   string `json:"changesOn"`
+	ChangesIn   int    `json:"changesIn"`
+	Usage       string `json:"usage"`
+	Used        int    `json:"used"`
+	Limit       int    `json:"limit"`
+	Measure     string `json:"measure"`
+}
+
+func (client LicenseClient) HaveGrace() bool {
+	return client.Status == "Grace"
+}
+
+func (client LicenseClient) IsExpired() bool {
+	return client.Status == "Expired"
+}
+
+func (client LicenseClient) IsExhausted() bool {
+	return client.Status == "Exhausted"
+}
+
+func (client LicenseClient) IsActive() bool {
+	return client.Status == "Active"
+}
+
+func (client LicenseClient) IsTrial() bool {
+	return client.LicenseType == "trial"
+}
+
+func (client LicenseClient) IsFree() bool {
+	return client.LicenseType == "free"
+}
+
+func (client LicenseClient) IsCommercial() bool {
+	return client.LicenseType == "commercial"
+}
+
+func (client LicenseClient) LicenseExpirationDate() time.Time {
+	expiresOn, err := time.Parse(time.RFC3339, client.ChangesOn)
+	if err != nil {
+		log.Fatal("Unknown expiration time received from the server: ", err)
```

Comment:
are you sure you want to use `log.Fatal` everywhere? this will exit the application. you can return the error